Change base docker image

aitorcas23 opened the pull request:
Change base python docker image because of incompatibilities in some packages in raspbian 11.

Couldn't execute systemctl reboot or poweroff because from inside the container.

aitorcas23 committed:
Change base docker image

Change base docker image because of incompatibility with new raspbian release.

aitorcas23 committed:
Rename variables and add logging

Rename shutdown host and port for consistency with other services.
Add logging to shutdown commands.

bodiroga commented on src/main.py:
```diff
@@ -8,8 +8,9 @@
 LOG_LEVEL = os.environ.get("LOG_LEVEL", logging.INFO)
-SHUTDOWN_EVENTS_HOST = "0.0.0.0"
-SHUTDOWN_EVENTS_PORT = int(os.environ.get("SHUTDOWN_EVENTS_PORT", 5558))
+SHUTDOWN_PORT = int(os.environ.get("SHUTDOWN_PORT", 5558))
```
@aitorcas23, ok, mejor nada que 'events', pero creo que lo idea sería que estas variables se llamaran "SHUTDOWN_SERVICE_HOST" y "SHUTDOWN_SERVICE_PORT"

aitorcas23 replied:
Ok, me parece bien, cambiaré todos los shutdown host y port de otros servicios también por consistencia.

bodiroga commented on src/main.py:
```diff
@@ -29,7 +30,7 @@ def signal_handler(sig, frame):
     shutdown_commands = ShutdownCommands()
 
-    server = CommunicationServer(shutdown_commands, host=SHUTDOWN_EVENTS_HOST, port=SHUTDOWN_EVENTS_PORT)
+    server = CommunicationServer(shutdown_commans, host=SHUTDOWN_HOST, port=SHUTDOWN_PORT)
```
```suggestion
    server = CommunicationServer(shutdown_commands, host=SHUTDOWN_HOST, port=SHUTDOWN_PORT)
```

@aitorcas23, se acaba de colar de nuevo el "commans", aunque esta vez la definición de la variable es correcta 😜

bodiroga commented on src/shutdown_commands.py:
```diff
@@ -1,12 +1,16 @@
         pass
 
     def shutdown(self):
+        logging.debug("shutdown command recieved.")
```
@aitorcas23, entiendo la idea, pero creo que el mensaje no es adecuado, lo siento. Esta clase no sabe nada sobre ningún comando/evento, su única función es apagar/reiniciar el dispositivo cuando llamen a sus funciones. El que sabe sobre los comandos es el CommunicationServer y ahí sí que imprimimos un mensaje en el log cuando se recibe un comando ```logger.debug(f"Command '{req_command}' has been requested")```. Si quieres añadir un mensaje aquí, yo optaría por:

```suggestion
        logging.debug("Shutting down the device.")
```

bodiroga commented on src/shutdown_commands.py:
```diff
@@ -1,12 +1,16 @@
         os.system("systemctl poweroff")
 
     def reboot(self):
+        logging.debug("reboot command recieved.")
```
```suggestion
        logging.debug("Rebooting the device.")
```

aitorcas23 committed:
Apply requested changes #3

- Rename shutdown host and port to shutdown service host and port.
- Fix spalling mistake on shutdown_commans to shutdown_commands.
- Rephrase shutdown commands logs.

aitorcas23 committed:
Add missing docker image to docker run example

bodiroga committed:
Fix tiny typo

bodiroga reviewed (approved):
LGTM! 👍🏻
